### PR TITLE
[tests-only][full-ci] add tests to remove access of shared resource to federated user

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds test to remove access of a shared resource to federated user.
``` feature
Scenario Outline: local user removes access of federated user from a resource
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/10073
- Bug Report: https://github.com/owncloud/ocis/issues/10213

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiOcm/share.feature [hunk 1]
```diff
@@ -753,3 +753,45 @@ Feature: an user shares resources using ScienceMesh application
...
+      | resource        | FOLDER   |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    When user "Alice" removes the access of user "Brian" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "REMOTE"
+    And for user "Brian" the space "Shares" should not contain these entries:
+      | FOLDER |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiOcm/share.feature:776]:
```suggestion
    Then for user "Brian" the space "Shares" should not contain these entries:
      | FOLDER |
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 2]
```diff
@@ -753,3 +753,45 @@ Feature: an user shares resources using ScienceMesh application
...
+      | space           | Personal |
+      | sharee          | Alice    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    And using server "REMOTE"
+    When user "Brian" removes the access of user "Alice" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "LOCAL"
+    And for user "Alice" the space "Shares" should not contain these entries:
+      | FOLDER |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiOcm/share.feature:797]:
```suggestion
    Then for user "Alice" the space "Shares" should not contain these entries:
      | FOLDER |
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 3]
```diff
@@ -753,3 +753,45 @@ Feature: an user shares resources using ScienceMesh application
...
+  @issue-10213
+  Scenario: local user deletes access of shared resource to federated user
+    Given using spaces DAV path
+    And using server "LOCAL"
+    And user "Alice" has created a folder "FOLDER" in space "Personal"
+    And "Alice" has created the federation share invitation
+    And using server "REMOTE"
+    And "Brian" has accepted invitation
+    And using server "LOCAL"
+    And user "Alice" has sent the following resource share invitation to federated user:
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:766]: this flow should also work
```suggestion
    And using server "REMOTE"
    And "Brian" has created the federation share invitation
    And using server "LOCAL"
    And "Alice" has accepted invitation
    And user "Alice" has created a folder "FOLDER" in space "Personal"
    And user "Alice" has sent the following resource share invitation to federated user:
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 4]
```diff
@@ -753,3 +753,45 @@ Feature: an user shares resources using ScienceMesh application
...
+      | resource        | FOLDER   |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    When user "Alice" removes the access of user "Brian" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "REMOTE"
+    And for user "Brian" the space "Shares" should not contain these entries:
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:775]: I have a doubt on this step. Could you check if this step works without deleting the share

Comment (prashant-gurung899) [reply tests/acceptance/features/apiOcm/share.feature:775]: yup the step is passing even without deleting the share.
Should we check the response or make another step to check for federated shares?

File: tests/acceptance/features/apiOcm/share.feature [hunk 5]
```diff
@@ -776,3 +776,75 @@ Feature: an user shares resources using ScienceMesh application
...
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    When user "Alice" removes the access of user "Brian" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "REMOTE"
+    And user "Brian" lists the shares shared with him without retry using the Graph API
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should not contain resource "FOLDER" with the following data:
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:799]: try to use this step. you might have to create for `not` assertion
`And user "Brian" should have a share "FOLDER" shared by user "Alice"`
```suggestion
```

**Comment by prashant-gurung899** [comment]
This PR is on hold because of bug; https://github.com/owncloud/ocis/issues/10689

**Comment by prashant-gurung899** [comment]
Moving it back to `in progress` from `blocked`

File: tests/acceptance/bootstrap/SharingNgContext.php [hunk 6]
```diff
@@ -1738,36 +1738,39 @@ public function userRemovesTheLastLinkShareOfSpaceUsingPermissionsEndpointOfGrap
...
 	 * @param bool $shouldExist
+	 * @param bool|null $isFederatedSHare
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 * @throws JsonException
 	 * @throws Exception
 	 */
-	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true): void {
+	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true, ?bool $isFederatedSHare= false): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SharingNgContext.php:1748]:
```suggestion
	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true, bool $federatedShare = false): void {
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 7]
```diff
@@ -1229,3 +1229,25 @@ Feature: an user shares resources using ScienceMesh application
...
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    When user "Alice" removes the access of user "Brian" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "REMOTE"
+    And user "Brian" lists the shares shared with him without retry using the Graph API
+    Then the HTTP status code should be "200"
+#    And user "Brian" should have a share "FOLDER" shared by user "Alice" from space "Personal"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiOcm/share.feature:1252]: please open up reason behind commenting step look @issue-10213 has been closed so this issue can't be reason

Comment (prashant-gurung899) [reply tests/acceptance/features/apiOcm/share.feature]: My bad! Forgot to remove the commented step.

File: tests/acceptance/bootstrap/SharingNgContext.php [hunk 8]
```diff
@@ -1808,6 +1811,21 @@ public function userShouldHaveShareSharedByUserFromSpace(string $sharee, string
...
+	 *
+	 * @param string $sharee
+	 * @param string $shouldOrNot
+	 * @param string $share
+	 * @param string $sharer
+	 * @param string $space
+	 *
+	 * @return void
+	 */
+	public function userShouldHaveFederatedShareSharedByUserFromSpace(string $sharee, string $shouldOrNot, string $share, string $sharer, string $space): void {
```
Comment (S-Panta) [inline tests/acceptance/bootstrap/SharingNgContext.php:1825]:
```suggestion
	public function userShouldOrShouldNotHaveFederatedShareSharedByUserFromSpace(string $sharee, string $shouldOrNot, string $share, string $sharer, string $space): void {
```
Just my suggestion

File: tests/acceptance/features/apiOcm/share.feature [hunk 9]
```diff
@@ -1229,3 +1229,24 @@ Feature: an user shares resources using ScienceMesh application
...
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    When user "Alice" removes the access of user "Brian" from resource "FOLDER" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    When using server "REMOTE"
+    And user "Brian" lists the shares shared with him without retry using the Graph API
+    Then the HTTP status code should be "200"
+    And user "Brian" should not have a federated share "FOLDER" shared by user "Alice" from space "Personal"
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:1252]:
```suggestion
    Then user "Brian" should not have a federated share "FOLDER" shared by user "Alice" from space "Personal"
```

File: tests/acceptance/bootstrap/SharingNgContext.php [hunk 10]
```diff
@@ -1738,36 +1738,39 @@ public function userRemovesTheLastLinkShareOfSpaceUsingPermissionsEndpointOfGrap
...
 	 * @param bool $shouldExist
+	 * @param bool $federatedShare
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 * @throws JsonException
 	 * @throws Exception
 	 */
-	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true): void {
+	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true, bool $federatedShare= false): void {
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/SharingNgContext.php:1748]:
```suggestion
	public function checkIfShareExists(string $share, string $sharee, string $sharer, string $space, bool $shouldExist = true, bool $federatedShare = false): void {
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 11]
```diff
@@ -1229,3 +1229,23 @@ Feature: an user shares resources using ScienceMesh application
...
+    And "Brian" has created the federation share invitation
+    And using server "LOCAL"
+    And "Alice" has accepted invitation
+    And user "Alice" has created a folder "FOLDER" in space "Personal"
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | FOLDER   |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiOcm/share.feature:1246]: better if your scenario cover all permissionsRole

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiOcm/share.feature [hunk 12]
```diff
@@ -1229,3 +1229,27 @@ Feature: an user shares resources using ScienceMesh application
...
+    And "Brian" has created the federation share invitation
+    And using server "LOCAL"
+    And "Alice" has accepted invitation
+    And user "Alice" has created a folder "FOLDER" in space "Personal"
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | FOLDER   |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | <permissionsRole>   |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiOcm/share.feature:1246]: format table

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Pull request merged**